src/components/tl/ucp/tl_ucp_ep.c

```diff
@@ -17,6 +23,13 @@ static inline ucc_status_t ucc_tl_ucp_connect_ep(ucc_tl_ucp_context_t *ctx,
     ep_params.field_mask = UCP_EP_PARAM_FIELD_REMOTE_ADDRESS;
     ep_params.address    = (ucp_address_t*)address->addr;
 
+    if (!UCC_TL_CTX_HAS_OOB(ctx)) {
+        ep_params.err_mode        = UCP_ERR_HANDLING_MODE_PEER;
+        ep_params.err_handler.cb  = ucc_tl_ucp_err_handler;
+        ep_params.err_handler.arg = NULL;
+        ep_params.field_mask     |= UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE |
+                                    UCP_EP_PARAM_FIELD_ERR_HANDLER;
+    }
```

Comment:
@vspetrov Should we warn the user about non-optimal behavior?

Reply:
@manjugv i think not, we might add some note in the manual or readme